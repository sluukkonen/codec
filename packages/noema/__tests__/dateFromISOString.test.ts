import { dateFromISOString } from "../src/index.js"
import { expectParseFailure, expectParseSuccess } from "./helpers.js"

describe("dateFromISOString", () => {
  test("should parse valid ISO dates", () => {
    const now = new Date()
    expectParseSuccess(dateFromISOString, now.toISOString(), now)
  })

  test("should reject non-strings", () => {
    expectParseFailure(dateFromISOString, null)
  })

  test("should reject invalid strings", () => {
    expectParseFailure(dateFromISOString, "asdf")
  })
})
