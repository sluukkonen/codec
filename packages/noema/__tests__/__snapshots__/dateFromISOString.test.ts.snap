// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dateFromISOString should reject invalid strings 1`] = `
[
  {
    "code": "invalid_iso_string",
    "path": [],
  },
]
`;

exports[`dateFromISOString should reject non-strings 1`] = `
[
  {
    "code": "invalid_type",
    "expected": "string",
    "path": [],
    "received": "null",
  },
]
`;
